### PR TITLE
VDS-546: View product list of the part

### PR DESCRIPTION
**Pull request by ilyawzrd**

### CODE REVIEW

File: src/VirtoCommerce.DemoSolutionFeaturesModule.Web/Scripts/blades/part-product-list.js [hunk 1]
```diff
@@ -0,0 +1,155 @@
...
+                name: "platform.commands.add", icon: 'fa fa-plus',
+                executeMethod: () => {},
+                canExecuteMethod: () => true
+            },
+            {
+                name: "platform.commands.delete", icon: 'fa fa-trash-o',
+                executeMethod: () => {
+                    const selectedNodesId = _.pluck($scope.gridApi.selection.getSelectedRows(), "id");
+                    if (_.some(selectedNodesId, (nodeId) => nodeId === blade.currentEntity.defaultItemId)) {
+                        bladeNavigationService.showConfirmationIfNeeded(true, true, blade, deleteProducts(selectedNodesId), ()=>{}, "demoSolutionFeaturesModule.dialogs.default-product-delete.title", "demoSolutionFeaturesModule.dialogs.default-product-delete.message");
```
Comment (pushnitsa) [inline src/VirtoCommerce.DemoSolutionFeaturesModule.Web/Scripts/blades/part-product-list.js]: I guess that pass successful callback should be wrapped with arrow function, like:
bladeNavigationService.showConfirmationIfNeeded(true, true, blade, **() => { deleteProducts(selectedNodesId); }**, ()=>{}, "demoSolutionFeaturesModule.dialogs.default-product-delete.title", "demoSolutionFeaturesModule.dialogs.default-product-delete.message");

File: src/VirtoCommerce.DemoSolutionFeaturesModule.Web/Scripts/blades/part-product-list.js [hunk 2]
```diff
@@ -0,0 +1,155 @@
...
+
+        $scope.setDefault = (contextMenuEntity) => {
+            const selectedNodeId = contextMenuEntity.id;
+            blade.currentEntity.defaultItemId = selectedNodeId;
+        };
+
+        $scope.delete = (contextMenuEntity) => {
+            const selectedNodeId = contextMenuEntity.id;
+            if (selectedNodeId === blade.currentEntity.defaultItemId) {
+                bladeNavigationService.showConfirmationIfNeeded(true, true, blade, deleteProduct(selectedNodeId), ()=>{}, "demoSolutionFeaturesModule.dialogs.default-product-delete.title", "demoSolutionFeaturesModule.dialogs.default-product-delete.message");
```
Comment (pushnitsa) [inline src/VirtoCommerce.DemoSolutionFeaturesModule.Web/Scripts/blades/part-product-list.js]: Wrap deleteProduct with arrow function

File: src/VirtoCommerce.DemoSolutionFeaturesModule.Web/Scripts/blades/part-product-list.tpl.html [hunk 3]
```diff
@@ -0,0 +1,109 @@
...
+             ui-grid-cell
+             data-target="prl_menu"
+             context-menu="grid.appScope.contextMenuEntity = row.entity">
+        </div>
+    </div>
+</script>
+
+<script type="text/ng-template" id="list-actions.cell.html">
+    <div class="ui-grid-actions" left-click-menu="grid.appScope.contextMenuEntity = row.entity" data-target="prl_menu">
+        <i class="fa fa-ellipsis-v"></i>
```
Comment (pushnitsa) [inline src/VirtoCommerce.DemoSolutionFeaturesModule.Web/Scripts/blades/part-product-list.tpl.html]: Add aria-hidden="true" attribute

File: src/VirtoCommerce.DemoSolutionFeaturesModule.Web/Scripts/blades/part-product-list.tpl.html [hunk 4]
```diff
@@ -0,0 +1,109 @@
...
+                    ui-grid-selection
+                    ui-grid-draggable-rows
+                    ui-grid-resize-columns
+                    ui-grid-move-columns
+                    ui-grid-pinning
+                    ui-grid-height>
+               </div>
+               <ul class="menu __context" role="menu" id="prl_menu">
+                   <li class="menu-item" ng-click='setDefault(contextMenuEntity)'>
+                       <i class="menu-ico fa fa-check"></i> {{'demoSolutionFeaturesModule.blades.part-product-list.context-menu.default' | translate}}
```
Comment (pushnitsa) [inline src/VirtoCommerce.DemoSolutionFeaturesModule.Web/Scripts/blades/part-product-list.tpl.html]: aria-hidden="true"

File: src/VirtoCommerce.DemoSolutionFeaturesModule.Web/Scripts/blades/part-product-list.tpl.html [hunk 5]
```diff
@@ -0,0 +1,109 @@
...
+                   </li>
+                   <li class="menu-item" ng-click='delete(contextMenuEntity)'>
+                       <i class="menu-ico fa fa-trash-o"></i> {{'platform.commands.delete' | translate}}
```
Comment (pushnitsa) [inline src/VirtoCommerce.DemoSolutionFeaturesModule.Web/Scripts/blades/part-product-list.tpl.html]: aria-hidden="true"

**Review by pushnitsa — CHANGES_REQUESTED** [review]

File: src/VirtoCommerce.DemoSolutionFeaturesModule.Web/Scripts/blades/part-product-list.tpl.html (hunk 4 above)
Comment (ilyawzrd) [reply src/VirtoCommerce.DemoSolutionFeaturesModule.Web/Scripts/blades/part-product-list.tpl.html]: done

File: src/VirtoCommerce.DemoSolutionFeaturesModule.Web/Scripts/blades/part-product-list.tpl.html (hunk 3 above)
Comment (ilyawzrd) [reply src/VirtoCommerce.DemoSolutionFeaturesModule.Web/Scripts/blades/part-product-list.tpl.html]: done

File: src/VirtoCommerce.DemoSolutionFeaturesModule.Web/Scripts/blades/part-product-list.tpl.html (hunk 5 above)
Comment (ilyawzrd) [reply src/VirtoCommerce.DemoSolutionFeaturesModule.Web/Scripts/blades/part-product-list.tpl.html]: done

File: src/VirtoCommerce.DemoSolutionFeaturesModule.Web/Scripts/blades/part-product-list.js (hunk 1 above)
Comment (ilyawzrd) [reply src/VirtoCommerce.DemoSolutionFeaturesModule.Web/Scripts/blades/part-product-list.js]: done

File: src/VirtoCommerce.DemoSolutionFeaturesModule.Web/Scripts/blades/part-product-list.js (hunk 2 above)
Comment (ilyawzrd) [reply src/VirtoCommerce.DemoSolutionFeaturesModule.Web/Scripts/blades/part-product-list.js]: done

File: src/VirtoCommerce.DemoSolutionFeaturesModule.Web/Scripts/blades/part-product-list.js [hunk 6]
```diff
@@ -0,0 +1,155 @@
...
+
+        filter.criteriaChanged = () => {
+            if ($scope.pageSettings.currentPage > 1) {
+                $scope.pageSettings.currentPage = 1;
+            } else {
+                blade.refresh();
+            }
+        };
+
+        $scope.canSave = () => {
```
Comment (pushnitsa) [inline src/VirtoCommerce.DemoSolutionFeaturesModule.Web/Scripts/blades/part-product-list.js]: () => isDirty();

File: src/VirtoCommerce.DemoSolutionFeaturesModule.Web/Scripts/blades/part-product-list.js [hunk 7]
```diff
@@ -0,0 +1,155 @@
...
+            _.each(selectedNodesId, (nodeId) => {
+                blade.currentEntity.partItems = _.filter(blade.currentEntity.partItems, (part) => {
+                    return part.itemId !== nodeId;
+                });
+            });
+            blade.refresh();
+        }
+
+        function deleteProduct(selectedNodeId) {
+            blade.currentEntity.partItems = _.filter(blade.currentEntity.partItems, (part) => {
```
Comment (pushnitsa) [inline src/VirtoCommerce.DemoSolutionFeaturesModule.Web/Scripts/blades/part-product-list.js]: (part) => part.itemId !== selectedNodeId

**Review by pushnitsa — CHANGES_REQUESTED** [review]

File: src/VirtoCommerce.DemoSolutionFeaturesModule.Web/Scripts/blades/part-product-list.js (hunk 6 above)
Comment (ilyawzrd) [reply src/VirtoCommerce.DemoSolutionFeaturesModule.Web/Scripts/blades/part-product-list.js]: done

File: src/VirtoCommerce.DemoSolutionFeaturesModule.Web/Scripts/blades/part-product-list.js (hunk 7 above)
Comment (ilyawzrd) [reply src/VirtoCommerce.DemoSolutionFeaturesModule.Web/Scripts/blades/part-product-list.js]: done

**Comment by ilyawzrd** [comment]
Снимок экрана 2021-01-14 в 16 26 47

**Review by pushnitsa — APPROVED** [review]

**Comment by ilyawzrd** [comment]
Closed. All changes now are part of https://github.com/VirtoCommerce/vc-module-demo-features/pull/51

**Pull request closed without merging**